src/commands/test.ts

```diff
@@ -24,15 +25,33 @@ export function sanitize(value?: string): string | undefined {
   return value?.replaceAll('"', '')
 }
 
-export function getTarget(environment: Environment, filter: string | undefined, { file, describe, test }: Options): Test[] {
-  const fqnByOptionalParameters = [file, describe, test].filter(Boolean).join('.')
-  const filterTest = sanitize(filter) ?? fqnByOptionalParameters ?? ''
-  const possibleTargets = environment.descendants.filter(is(Test))
+export function getTarget(environment: Environment, filter: string | undefined, options: Options): Test[] {
+  const possibleTargets = getBaseNode(environment, filter, options).descendants.filter(getTestFilter(filter, options))
   const onlyTarget = possibleTargets.find((test: Test) => test.isOnly)
   const testMatches = (filter: string) => (test: Test) => !filter || sanitize(test.fullyQualifiedName)!.includes(filter)
+  const filterTest = sanitize(filter) ?? ''
   return onlyTarget ? [onlyTarget] : possibleTargets.filter(testMatches(filterTest))
 }
 
+function getBaseNode(environment: Environment, filter: string | undefined, options: Options): Environment | Package | Describe {
+  if (filter) return environment
+
+  const { file, describe } = options
+  let nodeToFilter: Environment | Package | Describe = environment
+  if (file) {
+    nodeToFilter = environment.descendants.find(node => node.is(Package) && node.name === file) as Package | undefined ?? nodeToFilter
+  }
+  if (describe) {
+    nodeToFilter = environment.descendants.find(node => node.is(Describe) && node.name === `"${describe}"`) as Describe | undefined ?? nodeToFilter
+  }
+  return nodeToFilter
```

Comment:
Acá si se manda file y describe está ignorando el file, pero qué pasa si el nombre del describe se repite? (Similar a lo que estás intentando arreglar a nivel test).
También, si solamente se manda el nombre del archivo, se podría pensar qué pasa si hay 2 archivos con el mismo nombre en diferente carpetas... pero bueno.. por ahora no me calienta.

Pero propongo:
```suggestion
  const { file, describe } = options
  let nodeToFilter: Environment | Package | Describe = environment
  if (file) {
    nodeToFilter = nodeToFilter.descendants.find(node => node.is(Package) && node.name === file) as Package | undefined ?? nodeToFilter
  }
  if (describe) {
    nodeToFilter = nodeToFilter.descendants.find(node => node.is(Describe) && node.name === `"${describe}"`) as Describe | undefined ?? nodeToFilter
  }
  return nodeToFilter
```

Reply:
Tal vez estaría bueno tener un `Node.findByName<Describe>(name)`. Aunque este tipo de búsquedas son un toque peligrosas.